vms/evm/database/database_test.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"bytes"
+	"slices"
+	"testing"
+
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/ethdb/dbtest"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+
+	avalanchegodb "github.com/ava-labs/avalanchego/database"
+)
+
+// testDatabase wraps the production database with test-only snapshot functionality
+type testDatabase struct {
+	database
+}
+
+// NewSnapshot creates a test-only snapshot that captures the current database state.
+// This implementation is designed for test suite compliance only and is NOT suitable
+// for production use as it loads the entire database into memory.
+//
+// Creates a snapshot by iterating over the entire database and copying key-value pairs.
+func (db testDatabase) NewSnapshot() (ethdb.Snapshot, error) {
+
+	snapshotData := make(map[string][]byte)
+
+	iter := db.db.NewIterator()
+	defer iter.Release()
+
+	for iter.Next() {
+		key := iter.Key()
+		value := iter.Value()
+		keyCopy := make([]byte, len(key))
+		valueCopy := make([]byte, len(value))
+		copy(keyCopy, key)
+		copy(valueCopy, value)
+		snapshotData[string(keyCopy)] = valueCopy
+	}
+
+	if err := iter.Error(); err != nil {
+		return nil, err
+	}
+
+	return &testSnapshot{data: snapshotData}, nil
+}
+
+// testSnapshot implements ethdb.Snapshot by storing a copy of the database state.
+// This is a test-only implementation that loads the entire database into memory
+// for the purpose of satisfying the dbtest.TestDatabaseSuite interface requirements.
+// It is NOT suitable for production use due to memory and performance implications.
+type testSnapshot struct {
+	data map[string][]byte
+}
+
+func (s *testSnapshot) Get(key []byte) ([]byte, error) {
+	value, exists := s.data[string(key)]
+	if !exists {
+		return nil, avalanchegodb.ErrNotFound
```

Comment:
nit: It doesn't make a difference here, but we don't have to use avalanchego's database error here. Any error here works - we could just return some anonymous error like `errors.New("not found")`

Reply:
sure - guess we don't need the avalanchegodb